### DISCUSSION TITLE
Build failure on Manjaro Linux Pahvo

### DISCUSSION THREAD

**Discussion by Firestar-Reimu** [iraf-community/iraf#193] [Q&A] [answered]
```
getproc.c:24:13: note: ‘sprintf’ output between 7 and 262 bytes into a destination of size 256
   24 |             sprintf (fname, "/proc/%s", direntp->d_name);
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
make: *** [Makefile:19: sysgen] Error 1
```

It got this error when doing `make sysgen 2>&1 | tee build.log`, and the other commands are fine.

Full log can be seen at: https://pastebin.com/s4Dsq7Ud

Version: IRAF Snapshot 2.16.1+2021.06.14

**Comment by olebole** [comment, marked as answer]
This is the problem in your pastebin (lines 28ff):
```
In file included from irafpath.c:11:
/home/firestar/Downloads/iraf-main/include/iraf.h:1:1: error: expected identifier or ‘(’ before ‘.’ token
    1 | ../unix/hlib/libc/iraf.h
      | ^
```
The problem is that you messed up the symlinks and replaced them with text files in the download. `include/iraf.h` should be a symbolic link to  .`./unix/hlib/libc/iraf.h`, not a text file with that content. Please make sure that you either use the tar.gz file for the download.

**Reply by Firestar-Reimu** [reply]
Thanks! It is a miracle that after 7min21s, I have successfully compiled it **using the `tar.gz` file instead of the `zip` file (and I don't know why it has this difference)**. These are the test results:
```
$ ./test/run_tests
ecl.e: files.md ......
ecl.e: images.imcoords.md ...........
ecl.e: images.imfilter.md ..............
ecl.e: images.imfit.md ...
ecl.e: images.imgeom.md ........
ecl.e: images.immatch.md ..
ecl.e: lists.md .......
ecl.e: noao.astutil.md ........
ecl.e: noao.digiphot.photcal.md .
ecl.e: numerical-recipes.md ..........
ecl.e: os.md ...
ecl.e: programming.md ...........
ecl.e: README.md 
ecl.e: sys.vops.md .
ecl.e: testproc.md .........................
ecl.e: test-syntax.md ...xsa
ecl.e: utilities.nttools.md ..............
Test summary:     127 passed
      2 skipped
      1 xfailed
```
Does this seems OK? Can I delete the iraf-2.16.1-2021.06.14 folder from the `tar.gz` installer?

**Reply by olebole** [reply]
That is OK. "xfail" means "expected to fail".

**Comment by Firestar-Reimu** [comment]
Can I choose Konsole or Gnome-terminal or xfce-ternimal .etc when installing? How should I write? xgterm looks awful on my 2560x1600 screen

![Screenshot_20210914_193618](https://user-images.githubusercontent.com/65082432/133250654-821bae20-b5db-4db2-878d-5ff1a5c0ced4.png)

**Reply by olebole** [reply]
These consoles cannot handle the IRAF graphics, so it will work as long as you don't need it. For xgterm, you can try to use another (fixed, bitmap) font. Or you can use PyRAF which has its own graphics and can be used with any terminal.
